### ISSUE TITLE
[IOS] ability to customize BarButton with custom views ? 

### ISSUE THREAD

**Issue by a-eid** [grahammendick/navigation#496]
is there a way to customize the `BarButton` component to display customized views such as badges ? 

I wanna do something similar to this github notification icon but with a counter & it doesn't look possible 

![Screen Shot 2020-12-10 at 7 12 36 AM](https://user-images.githubusercontent.com/19273413/101724369-1626cd80-3ab7-11eb-9a5a-8be62479daaa.png)

**Comment by grahammendick** [comment]
There is a "badge" prop on the  `TabBarItem`. Is that what you're after or do you really want to show a badge on the `BarButton`?

**Comment by a-eid** [comment]
yeah I'd like to display a `badge` on the `BarButton` image, but a custom view would be ideal.

**Comment by a-eid** [comment]
for Example what if I wanna display an SVG icon or a vector Icon, from `react-native-vector-icon` for example ?

**Comment by grahammendick** [comment]
Cool, thanks for the interesting feature request

**Comment by cloudrco** [comment]
Would love this feature as well. Currently we’re using react navigation where we implemented custom view for the tab bar item. I think @a-eid is right that it’ll be more ideal for an option to have custom view aside from the current image prop https://github.com/grahammendick/navigation/blob/578c0a506f8480f43ff09a5807292d455649771a/NavigationReactNative/src/TabBarItem.tsx#L24

**Comment by grahammendick** [comment]
@cloudrco Thanks for your comment. 

The Navigation router only wraps the underlying native behaviour. It doesn’t try to add props for behaviour that isn’t supported natively. For example, the `UIBarButtonItem` supports a [customView property](https://developer.apple.com/documentation/uikit/uibarbuttonitem/1617129-customview?language=objc) so I’d like to support that in the Navigation router. But the `UITabBarItem` doesn’t have a customView property so I don’t plan on supporting it. How do you want to customize the view of the tab bar items?

**Comment by cloudrco** [comment]
Thanks for that explanation @grahammendick. Currently, we are evaluating which navigation to use within our app. Would love to use SVG icons instead of image within the TabBarItem. Is there any way we can implement this?

For example `<TabBarItem icon={() => <Icon />} />`

Also, this issue should be entitled for both Android and iOS. Shall I create a separate issue?

**Comment by grahammendick** [comment]
@cloudrco Good idea, please raise a separate issue so we can carry on the discussion

**Comment by grahammendick** [comment]
@a-eid The problem with the custom view is that the 'tintColor' prop won't be applied to it. Currently, if you set a 'tintColor' to 'red' then all the bar button images are colored red by iOS. That won't be the case with an image inside a custom view. So I don't think you'd want to use it for svg icons, for example. 

@julian-baumann I noticed you gave a 👍 . Could you explain how you plan on using it, please?

**Comment by julian-baumann** [comment]
Yeah, we once tried to add a dropdown menu to the "RightBar" in the NavigationBar on iOS. (With that package: https://www.npmjs.com/package/@react-native-menu/menu).
I now realize, that my issue maybe irrelevant to that current issue regarding the TabBar, not the Navigation Bar. I don't quite know how related they are.

**Comment by grahammendick** [comment]
@julian-baumann Why would you need a custom view? Couldn't you trigger the menu from a `BarButton` with an 'image' prop?

**Comment by julian-baumann** [comment]
I tried that, but it didn't work. And at that time that was pretty much the only package available supporting such a menu on iOS 14. It isn't important anymore, we did something else. Just named it as an example.

**Comment by grahammendick** [comment]
@a-eid I just [released navigation-react-native 7.4.0](https://github.com/grahammendick/navigation/releases/tag/v7.4.0-NavigationReactNative) with custom `BarButton` views on iOS

**Comment by grahammendick** [comment]
For example, 

```jsx
<LeftBar>
  <BarButton>
    <TouchableHighlight onPress={() => {}}>
      <Image />
    </TouchableHighlight>
  </BarButton>
</LeftBar>
```
